x/slashing/msg_server.go

```diff
@@ -0,0 +1,52 @@
+package slashing
+
+import (
+	"context"
+
+	addresscodec "cosmossdk.io/core/address"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/x/slashing/types"
+
+	"github.com/sedaprotocol/seda-chain/app/utils"
+)
+
+var _ types.MsgServer = msgServer{}
+
+type msgServer struct {
+	types.MsgServer
+	pubKeyKeeper          PubKeyKeeper
+	validatorAddressCodec addresscodec.Codec
+}
+
+type PubKeyKeeper interface {
+	HasRegisteredKey(ctx context.Context, validatorAddr sdk.ValAddress, index utils.SEDAKeyIndex) (bool, error)
+}
+
+func NewMsgServerImpl(sdkMsgServer types.MsgServer, pubKeyKeeper PubKeyKeeper, valAddrCdc addresscodec.Codec) types.MsgServer {
+	ms := &msgServer{
+		MsgServer:             sdkMsgServer,
+		pubKeyKeeper:          pubKeyKeeper,
+		validatorAddressCodec: valAddrCdc,
+	}
+	return ms
+}
+
+// Unjail overrides the default Unjail method to add an additional
+// check for registration of required public keys.
+func (m msgServer) Unjail(ctx context.Context, req *types.MsgUnjail) (*types.MsgUnjailResponse, error) {
```

Comment:
Should this not also check whether the scheme has been activated? I see a point in not having that check, will get more validators to register it ahead of time :D

Reply:
I didn't intend that 😂 Added the check